api/proto/teleport/legacy/types/types.proto

```diff
@@ -6213,6 +6215,125 @@ enum PluginStatusCode {
   SLACK_NOT_IN_CHANNEL = 10;
 }
 
+// PluginOktaStatusDetailsV1 contains the details for the running Okta plugin.
+message PluginOktaStatusDetailsV1 {
+  // SSODetails are status details relating to SSO.
+  PluginOktaStatusDetailsSSO sso_details = 2;
+
+  // AppGroupSyncDetails are status details relating to synchronizing apps and
+  // groups from Okta.
+  PluginOktaStatusDetailsAppGroupSync app_group_sync_details = 3;
+
+  // UsersSyncDetails are status details relating to synchronizing users from
+  // Okta.
+  PluginOktaStatusDetailsUsersSync users_sync_details = 4;
+
+  // ScimDetails are status details relating to SCIM integration with
+  // Okta.
+  PluginOktaStatusDetailsSCIM scim_details = 5;
+
+  // AccessListSyncDetails are status details relating to synchronizing access
+  // lists from Okta.
+  PluginOktaStatusDetailsAccessListsSync access_lists_sync_details = 6;
+}
+
+// PluginOktaStatusDetailsSSO are details related to the
+// current status of the Okta integration w/r/t SSO.
+message PluginOktaStatusDetailsSSO {
+  bool enabled = 1;
+  string app_id = 2;
+  string app_name = 3;
+}
+
+// PluginOktaStatusDetailsAppGroupSync are details related to the
+// current status of the Okta integration w/r/t application and group
+// sync.
+message PluginOktaStatusDetailsAppGroupSync {
+  // LastSuccessful is the date of the last successful run.
+  google.protobuf.Timestamp last_successful = 1 [
+    (gogoproto.stdtime) = true,
+    (gogoproto.nullable) = true,
+    (gogoproto.jsontag) = "last_successful"
+  ];
+
+  // LastFailed is the date of the last failed run.
+  google.protobuf.Timestamp last_failed = 2 [
```

Comment:
Hmm, why does these inner fields need a separate success and failed timestamps? Can't we deduce that from the `PluginStatusV1`'s  `last_sync_time time` and `PluginStatusCode`?.  

If the struct is to use elsewhere, outside of `PluginStatusV1` I would still suggest to have just a single timestamp field and instead have a sync status field `OktaPluginSyncStatusCode` to have sync run status.  Otherwise, we'll have to check both the time fields, compare which one is the latest to identify if the sync run was success or failure.

This also applies to `PluginOktaStatusDetailsUsersSync`, `PluginOktaStatusDetailsAccessListsSync` below. 

```
enum OktaPluginSyncStatusCode {
  RUNNING = 0;
  SUCCESS = 1;
  FAILED = 2;
}
```

Reply:
These are independent services that can fail independently, and we to be able to post the status of each part in the UI.

Reply:
...but yeah, having a single status value is still a good idea.

Reply:
Done